### PR TITLE
Deployment switcharoo

### PR DESCRIPTION
**Pull request by mgwalker**
# Still needs...
- [ ] preview deploy environment variables in CI/CD
- [ ] preview deployment cleanup spruced up

### This pull request changes...
- commits to master deploy to the `staging` environment
- tags on master deploy to the `production` environment
- deployments tag themselves in AWS with their environment to find them more easily
- renames environment variables to hopefully make them make semantic sense
   - `{ENVIRONMENT}_{TARGET}_{NAME}`
   - where `ENVIRONMENT` is `PRODUCTION`, `STAGING`, etc.
   - where `TARGET` is `API` or `WEB`
   - variable names related to AWS config start with `AWS_`
   - eg:
      - `PRODUCTION_API_AWS_AMI`
      - `PRODUCTION_API_PORT`
      - `STAGING_WEB_API_URL`

### This pull request is ready to merge when...
- [ ] This code has been reviewed by someone other than the original author
- [ ] documentation has been updated
  - [ ] CI/CD environment variables are updated
  - [ ] rules for deployment/environment are updated

### After merge
Remove these environment variables from CircleCI config:
- [ ] `AWS_PROD_API_AMI`
- [ ] `AWS_PROD_API_DATABASE_URL`
- [ ] `AWS_PROD_API_PORT`
- [ ] `AWS_PROD_API_REGION`
- [ ] `AWS_PROD_API_SECURITY_GROUP`
- [ ] `AWS_PROD_API_SUBNET`
- [ ] `AWS_PROD_API_TARGET_GROUP_ARN`
- [ ] `AWS_PROD_KEY_NAME`
- [ ] `AWS_PROD_S3_BUCKET`
- [ ] `AWS_PROD_S3_REGION`
- [ ] `PROD_API_URL`

### CODE REVIEW

**Comment by mgwalker** [comment]
Going to punt on updates to the preview deployments for now, since we can change those without affecting production - safe to leave them for later.

File: bin/preview-deploy/aws.sh [hunk 1]
```diff
@@ -15,15 +15,15 @@ function deployPreviewtoEC2() {
   configureAWS
 
   # Inject branch name into user data
-  configureUserData $2
+  configureUserData
```
Comment (hbillings) [inline bin/preview-deploy/aws.sh]: Isn't `configureUserData` still acting on the branch name? Does it not need to be passed in? I AM SO BAD AT BASH.

File: bin/preview-deploy/aws.sh [hunk 2]
```diff
@@ -108,10 +107,14 @@ function getPublicDNS() {
 #
 # $1 - list of instance ID to delete
 function terminateInstance() {
+  # This command can fail if AWS doesn't recognize the instance ID, but in that
+  # case we really, really don't care; so don't stop the script if it errors.
```
Comment (hbillings) [inline bin/preview-deploy/aws.sh]: 😂

File: bin/prod-deploy/aws.sh [hunk 3]
```diff
@@ -215,14 +230,13 @@ function waitForInstanceToBeReady() {
...
-#   AWS_PROD_API_TARGET_GROUP_ARN - The ARN of the target group to remove the
-#     instance from.
+# Expects global variables:
+#   API_PORT - Port the EC2 instance is listening on
+#   AWS_TARGET_GROUP - The ARN of the target group to remove the instance from.
 function waitForTargetToBeHealthy() {
   HEALTH_CHECK_COUNT=1
   TARGET_HEALTH=$(getTargetHealth $1)
-  while [[ "$TARGET_HEALTH" != "healthy" && "HEALTH_CHECK_COUNT" -lt 10 ]]; do
+  while [[ "$TARGET_HEALTH" != "healthy" && "HEALTH_CHECK_COUNT" -lt 60 ]]; do
```
Comment (hbillings) [inline bin/prod-deploy/aws.sh]: Guessing that bumping up the health check count gives us more retries?

File: bin/preview-deploy/aws.sh (hunk 1 above)
Comment (mgwalker) [reply bin/preview-deploy/aws.sh]: Ah, bad comment! You're right, it is, but it's getting it from the `$BRANCH` local global (I don't know what I'm saying either), so it doesn't have to be passed in as an explicit argument anymore. Going to push an updated comment.

File: bin/prod-deploy/aws.sh (hunk 3 above)
Comment (mgwalker) [reply bin/prod-deploy/aws.sh]: Yeah. This is probably overkill, but it turns out that *every now and then* it takes more than 10 minutes for a machine to start responding. No idea why. Probably gremlins.

Comment (hbillings) [reply bin/prod-deploy/aws.sh]: satan

Comment (mgwalker) [reply bin/prod-deploy/aws.sh]: well damn

**Pull request merged**